client/my-sites/guided-transfer/host-credentials-page/index.jsx

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Bluehost from './bluehost';
+import SiteGround from './siteground';
+import SectionHeader from 'components/section-header';
+
+class HostCredentialsPage extends Component {
+	static propTypes = {
+		hostSlug: PropTypes.string.isRequired,
+	};
+
+	state = { fieldValues: {} };
```

Comment:
How about instead of using `state` we use `ref`. So you'd end up with some class methods here like
```js
    setUsernameRef = ref => this._usernameField = ref;
    setEmailRef = ref => this._emailField = ref;
    setPasswordRef = ref => this._passwordField = ref;
    setDestinationRef = ref => this._destinationField = ref;
```
Pass the `set...Ref` methods down as props into the fields that are in `fields.jsx`
then in your `submit` method below you'll be able to do 
```js
    submit = () => {
        const username = this._usernameField.value;
        const password = this._passwordField.value;
        const email = this._emailField.value;
        const destination = this._destinationField.value;
        // Trigger the redux API call
    }
```

I like this approach because:
 1. No obscure functions like `setFieldValue` and `onFieldChange` that set dynamic values in the state tree
 2. No more rendering with every keystroke.
 3. Its not using state anymore.